File: android/java/chartboostkore/ChartboostKore.java
package chartboostkore;

import com.chartboost.sdk.Libraries.CBLogging.Level;
import com.chartboost.sdk.Model.CBError.CBClickError;
import com.chartboost.sdk.Model.CBError.CBImpressionError;
import com.chartboost.sdk.Tracking.CBAnalytics;
import com.chartboost.sdk.CBLocation;
import com.chartboost.sdk.CBImpressionActivity;
import com.chartboost.sdk.Chartboost;
import com.chartboost.sdk.ChartboostDelegate;

public class ChartboostKore {
    
	public static void init() {
		//Chartboost.startWithAppId(this, appId, appSignature);
	}

	public static void showInterstitial() {	
		Chartboost.showInterstitial(CBLocation.LOCATION_DEFAULT);
	}

	public static void cacheInterstitial() {
		Chartboost.cacheInterstitial(CBLocation.LOCATION_DEFAULT);
	}

	public static boolean hasInterstitial() {
		return Chartboost.hasInterstitial(CBLocation.LOCATION_DEFAULT);
	}
}

File: android/chartboostkore/ChartboostKore.cpp
#include <ChartboostKore.h>

#include <Kore/Android.h>

namespace ChartboostKore {

	int init(const char* appId, const char* appSignature) {
		

		return 0;
	}

	void showInterstitial() {
		JNIEnv* env;
		KoreAndroid::getActivity()->vm->AttachCurrentThread(&env, NULL);

		jclass cls = KoreAndroid::findClass(env, "chartboostkore.ChartboostKore");

        jmethodID methodId = env->GetStaticMethodID(cls, "showInterstitial", "()V");

        env->CallStaticVoidMethod(cls, methodId);

        KoreAndroid::getActivity()->vm->DetachCurrentThread();
	}

	void cacheInterstitial() {
		JNIEnv* env;
		KoreAndroid::getActivity()->vm->AttachCurrentThread(&env, NULL);

		jclass cls = KoreAndroid::findClass(env, "chartboostkore.ChartboostKore");

        jmethodID methodId = env->GetStaticMethodID(cls, "cacheInterstitial", "()V");

        env->CallStaticVoidMethod(cls, methodId);

        KoreAndroid::getActivity()->vm->DetachCurrentThread();
	}

	bool hasInterstitial() {
		JNIEnv* env;
		KoreAndroid::getActivity()->vm->AttachCurrentThread(&env, NULL);

		jclass cls = KoreAndroid::findClass(env, "chartboostkore.ChartboostKore");

        jmethodID methodId = env->GetStaticMethodID(cls, "hasInterstitial", "()Z");

        bool result = env->CallStaticBooleanMethod(cls, methodId);
        
        KoreAndroid::getActivity()->vm->DetachCurrentThread();

        return result;
	}

	void showRewardedVideo() {

	}

	void cacheRewardedVideo() {

	}
}
